buffer/src/main/java/io/netty/buffer/UnpooledDirectByteBuf.java

```diff
@@ -341,19 +341,18 @@ public ByteBuf getBytes(int index, ByteBuffer dst) {
     }
 
     private void getBytes(int index, ByteBuffer dst, boolean internal) {
-        checkIndex(index);
+        checkIndex(index, dst.remaining());
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Nullcheck of dst at line 345 of value previously dereferenced in io.netty.buffer.UnpooledDirectByteBuf.getBytes(int, ByteBuffer, boolean) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=findbugs%3ARCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE)